HLTrigger/special/src/HLTRechitsToDigis.cc

```diff
@@ -0,0 +1,204 @@
+//
+// Package:    HLTrigger/special
+// Class:      HLTRechitsToDigis
+// 
+/**\class HLTRechitsToDigis HLTRechitsToDigis.cc HLTrigger/special/plugins/HLTRechitsToDigis.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Joshua Robert Hardenbrook
+//         Created:  Fri, 20 Feb 2015 15:51:36 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+#include "CondFormats/EcalObjects/interface/EcalChannelStatus.h"
+#include "CondFormats/DataRecord/interface/EcalChannelStatusRcd.h"
+
+//
+// class declaration
+//
+
+class HLTRechitsToDigis : public edm::stream::EDProducer<> {
+public:
+  explicit HLTRechitsToDigis(const edm::ParameterSet&);
+  ~HLTRechitsToDigis();
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  virtual void produce( edm::Event&, edm::EventSetup const&) override;
+      
+  // ----------member data ---------------------------  
+  // tokens for the digi and rechits for matching
+  edm::EDGetTokenT<EBDigiCollection> digisEBInToken_;
+  edm::EDGetTokenT<EEDigiCollection> digisEEInToken_;
+  edm::EDGetTokenT<EcalRecHitCollection> recHitsToken_;
+  
+  // input tags
+  edm::InputTag digisIn_;
+  edm::InputTag recHits_;
+  
+  // string for the produced digi collection
+  std::string digisOut_;
+  uint32_t regionID_;
+};
+//
+
+//
+// constructors and destructor
+//
+HLTRechitsToDigis::HLTRechitsToDigis(const edm::ParameterSet& iConfig)
+{
+  //region to do rechit digi matching
+  regionID_ = iConfig.getParameter<uint32_t> ("regionID_1EB_2EE");
```

Comment:
No, just no. Please use simply "region" as the parameter name.

You can use strings ("EE" / "EB" or "endcap" / "barrel" or whatever you prefer) and covert them integers (better, an enum) in a helper function.
Or, plain integers as you are using, and add a comment in the cfi file explaining the meaning.